### PR TITLE
Upgrade to v0.11.1 spec

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Upgrade Teku to the v0.11.1 spec, excluding network changes.

Note that fork choice changes haven't been started yet and validator changes are incomplete.

### CODE REVIEW

File: ethereum/core/src/main/java/tech/pegasys/artemis/core/EpochProcessorUtil.java [hunk 1]
```diff
@@ -612,22 +610,22 @@ public static void process_final_updates(MutableBeaconState state) {
...
-      long HALF_INCREMENT = Constants.EFFECTIVE_BALANCE_INCREMENT / 2;
-      if (balance.compareTo(validator.getEffective_balance()) < 0
-          || validator
-                  .getEffective_balance()
-                  .plus(UnsignedLong.valueOf(3 * HALF_INCREMENT))
-                  .compareTo(balance)
-              < 0) {
 
+      final UnsignedLong hysteresis_increment =
+          EFFECTIVE_BALANCE_INCREMENT.dividedBy(HYSTERESIS_QUOTIENT);
```
Comment (cemozerr) [inline ethereum/core/src/main/java/tech/pegasys/artemis/core/EpochProcessorUtil.java:615]: Seems like these 3 variables could go into the Constants file

File: ethereum/core/src/main/java/tech/pegasys/artemis/core/Signer.java [hunk 2]
```diff
@@ -35,29 +36,49 @@ public Signer(final MessageSignerService signerService) {
...
 
-  public SafeFuture<BLSSignature> createRandaoReveal(final UnsignedLong epoch, final Fork fork) {
-    Bytes domain = get_domain(Constants.DOMAIN_RANDAO, epoch, fork);
+  public SafeFuture<BLSSignature> createRandaoReveal(
+      final UnsignedLong epoch, final ForkInfo forkIfno) {
+    Bytes domain =
+        get_domain(
+            Constants.DOMAIN_RANDAO,
+            epoch,
+            forkIfno.getFork(),
```
Comment (cemozerr) [inline ethereum/core/src/main/java/tech/pegasys/artemis/core/Signer.java]: nit:
```suggestion
            forkInfo.getFork(),
```

**Review by cemozerr — APPROVED** [review]
The changes here look good to me.

File: ethereum/core/src/main/java/tech/pegasys/artemis/core/EpochProcessorUtil.java (hunk 1 above)
Comment (ajsutton) [reply ethereum/core/src/main/java/tech/pegasys/artemis/core/EpochProcessorUtil.java:615]: If the constants were actually constants yes, but we'd need to be able to reset them after loading from the constants yaml file.

**Pull request merged**